lib/print.js

```diff
@@ -62,7 +62,21 @@ const printApply = (node, parent) => {
         }
         return result;
     } else if (op === 'pow') {
-        return `${print(args[0], node)}^${print(args[1], node)}`;
+        const exponent = node.args[1];
+        let bool = false;
+        // check if exponent has multiple terms and if the exponent
+        // contains a div/mul operation
+        // e.g x^((x + 1)/(2 + 2))
+        if (exponent.args){
+            if (exponent.op == 'div' || exponent.op == 'mul'){
+                bool = true;
+            }
+        }
+        if (bool) {
```

Comment:
Can you change this to `if (isMul(exponent) || isDiv(exponent) {` and define those helper functions as necessary.  Then you can get rid of the comment above b/c the intent is clear from the code.